tests/_conftest/devnet/faucet_contract.py

```diff
@@ -1,46 +1,21 @@
-from starknet_py.constants import DEVNET_FEE_CONTRACT_ADDRESS
-from starknet_py.contract import Contract
-from starknet_py.net import AccountClient
+import requests
 
 from protostar.starknet_gateway import Wei
 
 
 class FaucetContract:
     """Contract for prefunding deployments"""
 
-    def __init__(self, predeployed_account_client: AccountClient) -> None:
-        self._predeployed_account_client = predeployed_account_client
+    def __init__(self, devnet_gateway_url: str) -> None:
```

Comment:
I think in this form, this class could easily be inlined. Right now this has boiled to being simply a function which does very simple post request to devnet. Other tests here simply call `requests.post` directly.

Reply:
`devnet_gateway_url` is a dependency, whereas `recipient`, `amount` are parameters. Dependencies are usually provided in a different place compared to the place where methods are called. Such approach makes code less fragile (notice I didn't have to modify `DevnetAccountPreparator` despite changes in dependencies).

> Other tests here simply call requests.post directly.

There's one test that sends the request directly. Why should "test_account_with_tx_version_0" care about sending `"lite": True` to devnet?

Reply:
`lite` parameter is actually quite important because it determines whether a minting transaction is created, which may be needed in the test or not

Reply:
and notice, that now instead of passing `devnet_gateway_url` you simply pass `FaucetContract` as dependency everywhere

Reply:
> lite parameter is actually quite important because it determines whether a minting transaction is created, which may be needed in the test or not

I know it's important. I used it as an example of polluting test cases with implementation details. `lite` doesn't tell anything unless you read devnet docs.

> and notice, that now instead of passing devnet_gateway_url you simply pass FaucetContract as dependency everywhere

Using `devnet_gateway_url` everywhere encourages developers to duplicate the code intentionally or not and producing tests that are harder to read and maintain (unless is used as an argument for `ProtostarFixture`).

`FaucetContract` is passed where is needed. `DevnetFixture` will be the gateway to devnet as `ProtostarFixture` is a gateway to Protostar.

Reply:
do as you wish